docs/book/mlops-stacks/model-deployers/kserve.md

```diff
@@ -192,13 +192,90 @@ pytorch_model_deployer = kserve_model_deployer_step(
 {% endhint %}
 
-A concrete example of using the KServe Model Deployer can be found
-[here](https://github.com/zenml-io/zenml/tree/main/examples/kserve_deployment).
+## Custom Model Deployment
 
-For more information and a full list of configurable attributes of the KServe Model Deployer, check out the 
-[API Docs](https://apidocs.zenml.io/latest/api_docs/integrations/#zenml.integrations.kserve.model_deployers).
+While KServe is a good fit for most use cases with the built-in model servers, it is not always the best fit for your custom model deployment use case. For that reason KServe allow you to create your own model server using KServe ModelServer API where you can customize the predict, the pre- and post-processing functions.
+With ZenML's KServe Integration, you can create your own custom model
+deployment code by creating a custom predict function that will be passed
+to a custom deployment step responsible for preparing a docker image for the model server.
+
+This `custom_predict` function should be getting the model and the input data as arguments and return the output data. ZenML will take care of loading the model
+into memory, starting the KServe `ModelServer` that will be responsible for serving the model, and running the predict function.
+
+```python
+def pre_process(tensor: torch.Tensor) -> dict:
+    """Pre process the data to be used for prediction."""
+    pass
+
+
+def post_process(prediction: torch.Tensor) -> str:
+    """Pre process the data"""
+    pass
+
+
+def custom_predict(
+    model: Any,
+    request: dict,
+) -> dict:
+    """Custom Prediction function.
+
+    The custom predict function is the core of the custom deployment, the function
+    is called by the custom deployment class defined for the serving tool.
+    The current implementation requires the function to get the model loaded in the memory and
+    a request with the data to predict.
+
+    Args:
+        model (Any): The model to use for prediction.
+        request: The prediction response of the model is an Array_Like object.
+    Returns:
+        The prediction in an Array_Like. (e.g: np.ndarray, List[Any], str, bytes, Dict[str, Any])
+    """
+    pass
+```
+
+Then this custom predict function `path` can be passed to the custom deployment parameters.
+
+```python
+kserve_pytorch_custom_deployment = kserve_custom_model_deployer_step(
+    config=KServeDeployerStepConfig(
+        service_config=KServeDeploymentConfig(
+            model_name="kserve-pytorch-custom-model",
+            replicas=1,
+            predictor="custom",
+            resources={"requests": {"cpu": "200m", "memory": "500m"}},
+        ),
+        timeout=240,
+        custom_deploy_parameters=CustomDeployParameters(
+            predict_function="kserve_pytorch.steps.pytorch_custom_deploy_code.custom_predict"
```

Comment:
as a general question wouldnt it have made sense to pass this function as a function itself rather than the string?

Reply:
can we pass a callable into step config? I thought the last time I tested this it was failing. I agree that passing the function would be much better as we can avoid importing issues in a different environment. Tho the current implementation runs a test on the path to check that it exists and that it's a callable function before running or deploying.